x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryRewriter.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.query;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.CombinedFieldsQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.index.query.TermsQueryBuilder;
+import org.elasticsearch.xpack.relevancesearch.relevance.QueryConfiguration;
+import org.elasticsearch.xpack.relevancesearch.relevance.curations.CurationSettings;
+import org.elasticsearch.xpack.relevancesearch.relevance.curations.CurationsService;
+import org.elasticsearch.xpack.relevancesearch.relevance.settings.RelevanceSettings;
+import org.elasticsearch.xpack.relevancesearch.relevance.settings.RelevanceSettingsService;
+import org.elasticsearch.xpack.searchbusinessrules.PinnedQueryBuilder;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class RelevanceMatchQueryRewriter {
```

Comment:
Rewriter class, it uses the Builder as a POJO to get the query, settings id, etc and generate the query.